nomad/util.go

```diff
@@ -196,6 +196,34 @@ func shuffleStrings(list []string) {
 	}
 }
 
+// partitionAll splits a slice of strings into a slice of slices of strings, each with a max
+// size of `size`. All entries from the original slice are preserved. The last slice may be
+// smaller than `size`. The input slice is unmodified
+func partitionAll(size int, xs []string) [][]string {
+	out := make([][]string, 0)
```

Comment:
nit: You don't need to allocate slices if you're `append`ing into them. This could be `var result [][]string`.